Microsoft.HealthVault.Fhir/FhirExtensions/ObservationExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Hl7.Fhir.Model;
+using Microsoft.HealthVault.Thing;
+
+namespace Microsoft.HealthVault.Fhir.FhirExtensions
+{
+    public static class ObservationExtensions
+    {
+        public static void AddCommonData(this Observation observation, CommonItemData commonData)
+        {
+            if (commonData != null)
+            {
+                observation.AddNoteAsText(commonData.Note);
+
+                observation.AddSourceAsDevice(commonData.Source);
+
+                observation.AddRelated(commonData.RelatedItems);
+            }
+        }
+
+        public static void AddRelated(this Observation observation, ICollection<ThingRelationship> relatedItems)
+        {
+            if (relatedItems != null && relatedItems.Any())
+            {
+                observation.Related = new List<Observation.RelatedComponent>();
+                foreach (var item in relatedItems)
+                {
+                    if (item.ItemKey != null)
+                    {
+                        observation.Related.Add(new Observation.RelatedComponent() { Target = new ResourceReference($"observation/{item.ItemKey.Id}") });
```

Comment:
The concept of relationship in FHIR and HealthVault seems to be incompatible. FHIR Observation expects another _Observation_, _QuestionnaireResponse_ or _Sequence_ in related target, while Healthvault allows any _Thing_. This would require further discussion.

Again, this was refactoring of existing code to a function.